core/frontend/src/components/bridges/BridgetUpdater.vue

```diff
@@ -20,12 +20,21 @@ import { callPeriodically } from '@/utils/helper_functions'
 
 export default Vue.extend({
   name: 'BridgetUpdater',
+  data() {
+    return {
+      prefetched_bridges: false,
+      prefeteched_serial_ports: false,
+    }
+  },
   mounted() {
     callPeriodically(this.fetchAvailableBridges, 5000)
     callPeriodically(this.fetchAvailableSerialPorts, 5000)
   },
   methods: {
     async fetchAvailableBridges(): Promise<void> {
+      if (this.prefetched_bridges && !bridget.should_fetch) {
```

Comment:
It would be nice that we could encapsulate the should_fetch logic inside a component that works with axios to be used in the others components without copy/paste or duplicated code.